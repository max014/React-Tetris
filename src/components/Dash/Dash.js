import React from 'react';
import NumberHolder from '../NumberHolder/NumberHolder';
import styles from './Dash.module.css';
import { connect } from 'react-redux';

export const Dash = (props) => (
	<div className={styles.Dash}>
		<NumberHolder title="Level" number={props.level} />
		<NumberHolder title="Score" number={props.score} />
		<NumberHolder title="Lines" number={props.lines} />
	</div>
);

const mapStateToProps = state => {
    return {
        level: state.level,
        score: state.score,
        lines: state.lines
    };
}

export default connect(mapStateToProps)(Dash);