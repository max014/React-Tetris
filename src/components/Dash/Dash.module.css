.Dash{
	position: absolute;
	left: 110%;
	top: 5%;
	width: 25vw;
	box-sizing: border-box;
	padding-left: 20px;
	border: 4px solid white;
	background: linear-gradient(to left, #111, #222);
}