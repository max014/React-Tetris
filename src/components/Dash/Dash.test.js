import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Dash } from './Dash';

describe('Dash', () => {
    const props = {
        level: 1,
        score: 100,
        lines: 15
    };
    const dash = shallow(<Dash {...props}/>);
    it('renders properly', () => {
        expect(toJson(dash)).toMatchSnapshot();
    });
});