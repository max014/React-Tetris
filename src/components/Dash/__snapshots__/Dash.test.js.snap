// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dash renders properly 1`] = `
<div
  className="Dash"
>
  <NumberHolder
    number={1}
    title="Level"
  />
  <NumberHolder
    number={100}
    title="Score"
  />
  <NumberHolder
    number={15}
    title="Lines"
  />
</div>
`;
